### PR TITLE
Dev read multiprocessing

### PR DESCRIPTION
**Pull request by jlplenio**
multicore read implemented. If tests are ok, we have to check the executables.
There can be complications with multiprocessing and pyinstaller.

### CODE REVIEW

**Comment by jlplenio** [comment]
cpu_count does count logical core!

bovheat.exe fails when reading files. Will try this:
https://docs.python.org/2/library/multiprocessing.html#multiprocessing.freeze_support

**Comment by jlplenio** [comment]
Win executable works perfectly now with multiprocessing. Ready to merge.

**Comment by ndevln** [comment]
Machst du noch eine Flag mit dem ich das steuern kann?

File: bovheat_src/bh_input.py [hunk 1]
```diff
@@ -68,7 +67,7 @@ def get_args():
         "--cores",
         type=int,
         default="0",
-        help="specify amount of cores to use, default results in logical core count -1",
+        help="specify amount of cores to use, default 0 results in logical core count -1",
```
Comment (ndevln) [inline bovheat_src/bh_input.py:70]: "logical core count of your processor"?

File: bovheat_src/bh_input.py [hunk 2]
```diff
@@ -198,14 +196,20 @@ def get_source_data(language, core_count=0, relative_path=""):
...
-    if core_count == 0:
-        read_cores = multiprocessing.cpu_count() - 1
-    else:
-        read_cores = core_count
+    print(len(file_list), "files found.", end="")
 
-    print(f"{len(file_list)} files found. Reading with {read_cores} core(s) ...")
-    with multiprocessing.Pool(processes=read_cores) as pool:
-        df_list = pool.starmap(read_clean_file, file_list)
+    if core_count == 1:  # do not use multiprocessing
```
Comment (ndevln) [inline bovheat_src/bh_input.py:201]: Maybe this should be stated in the help text.

File: bovheat_src/bh_input.py (hunk 1 above)
Comment (ndevln) [reply bovheat_src/bh_input.py:70]: or CPU

File: tests/unit/test_read_sourcedata_and_clean.py [hunk 3]
```diff
@@ -2,19 +2,25 @@
 import pytest
 from bovheat_src import bh_input, bovheat
 
+# runs tests with multiprocessing. 0: auto-core, 1: disabled, 2: fixed 2 cores
+@pytest.fixture(params=[0, 1, 2])
```
Comment (ndevln) [inline tests/unit/test_read_sourcedata_and_clean.py:6]: Bonus points 👍

File: bovheat_src/bh_output.py [hunk 4]
```diff
@@ -64,6 +64,12 @@ def write_pdf(heats_df, sections_df, threshold, filename):
 
     pdf_file = PdfPages(filename)  # Start PDF file
 
+    # # example to parallelize group processing
```
[the remark this answers is not in the document]
Comment (ndevln) [reply bovheat_src/bh_output.py:67]: Maybe remove?

[the remark this answers is not in the document]
Comment (jlplenio) [reply bovheat_src/bh_output.py:67]: idea for pdf multiprocessing.

File: tests/unit/test_read_sourcedata_and_clean.py (hunk 3 above)
Comment (jlplenio) [reply tests/unit/test_read_sourcedata_and_clean.py:6]: quite yummy.

**Pull request merged**